docs/structured-streaming-programming-guide.md

```diff
@@ -1134,6 +1134,73 @@ sessionizedCounts = events \
 </div>
 </div>
 
+Instead of static value, we can also provide an expression to specify gap duration dynamically
+based on the input row. Note that the rows with negative or zero gap duration will be filtered
+out from the aggregation.
+
+<div class="codetabs">
+<div data-lang="scala"  markdown="1">
+
+{% highlight scala %}
+import spark.implicits._
+
+val events = ... // streaming DataFrame of schema { timestamp: Timestamp, userId: String }
+
+val sessionWindow = SessionWindow($"timestamp".expr, when($"userId" === "user1", "5 seconds")
```

Comment:
Why not adding override method of `session_window` into functions.scala? It sounds inconvenient if I need to remember dynamic gap duration is only supported with `SessionWindow` expression.

Reply:
Yea, also had the same thought. Let me add another `session_window` function.